ReactiveCocoa/Swift/Signal.swift

```diff
@@ -821,6 +821,13 @@ extension SignalType {
 			.map { $0.0 }
 	}
 
+	/// Forwards events from `self` until `lifetime` ends, at which point the
+	/// returned signal will complete.
+	@warn_unused_result(message="Did you forget to call `observe` on the signal?")
+	public func takeWithin(lifetime: Lifetime) -> Signal<Value, Error> {
```

Comment:
> Why not make this take an NSObject instead, and make Lifetime an implementation detail?

I think the idea is to make this usable with non-`NSObject` classes.

How about naming this `take(during lifetime: Lifetime)`. That seems like it would read a little more naturally.

```swift
.take(during: myObject.lifetime)
```